platforms/linux/flags-aarch64.cmake

```diff
@@ -0,0 +1,6 @@
+# see https://gcc.gnu.org/onlinedocs/gcc/AArch64-Options.html
+if(ENABLE_BF16)
+  set(OCV_FLAGS "-march=armv8.6-a")
+elseif(ENABLE_FP16 OR ENABLE_DOTPROD)
+  set(OCV_FLAGS "-march=armv8.4-a")
```

Comment:
Google uses `-march=armv8.2-a+bf16` for dispatched code:

https://github.com/google/XNNPACK/blob/f416ab3d2781c8eb53eaa999239138296146b5e0/CMakeLists.txt#L836

Why do we need `8.4`?

Reply:
I used this table from GCC documentation to establish baselines with assumption that actual hardware also lines up with it and there are no platforms which have `dotprod` on top of `armv8.2-a`, plus an assumption that compiler can optimize code better for such "baseline" arch.
```
arch value	Architecture	Includes by default
‘armv8-a’	Armv8-A	‘+fp’, ‘+simd’
‘armv8.1-a’	Armv8.1-A	‘armv8-a’, ‘+crc’, ‘+lse’, ‘+rdma’
‘armv8.2-a’	Armv8.2-A	‘armv8.1-a’
‘armv8.3-a’	Armv8.3-A	‘armv8.2-a’, ‘+pauth’
‘armv8.4-a’	Armv8.4-A	‘armv8.3-a’, ‘+flagm’, ‘+fp16fml’, ‘+dotprod’
‘armv8.5-a’	Armv8.5-A	‘armv8.4-a’, ‘+sb’, ‘+ssbs’, ‘+predres’
‘armv8.6-a’	Armv8.6-A	‘armv8.5-a’, ‘+bf16’, ‘+i8mm’
```
Should I change it back to `armv8.2-a+...`?

Reply:
Changed it to use `armv8.2-a+...` combination with feature flags. It became slightly more complicated, but I believe it's OK. Please let me know if we need to revert it.

Reply:
Also added to the toolchain ability to disable NEON for AArch64 by explicitly setting `ENABLE_NEON` to `OFF`.